File: 벤치마크/rust/fileio.rs
use std::env;
use std::fs;
use std::process;
use std::time::Instant;

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        eprintln!("Usage: fileio <filepath>");
        process::exit(1);
    }

    let data = match fs::read(&args[1]) {
        Ok(d) => d,
        Err(e) => {
            eprintln!("Cannot open {}: {}", args[1], e);
            process::exit(1);
        }
    };

    let start = Instant::now();

    let mut words: i64 = 0;
    let mut in_word = false;
    for &b in &data {
        if b == b' ' || b == b'\t' || b == b'\n' || b == b'\r' || b == 0x0b || b == 0x0c {
            in_word = false;
        } else {
            if !in_word {
                words += 1;
                in_word = true;
            }
        }
    }

    let ms = start.elapsed().as_secs_f64() * 1000.0;

    println!("words: {}", words);
    println!("RESULT:{:.3}", ms);
}

File: bench/java/FileIO.java
import java.nio.file.Files;
import java.nio.file.Paths;

public class FileIO {
    public static void main(String[] args) throws Exception {
        if (args.length < 1) {
            System.err.println("Usage: java FileIO <filepath>");
            System.exit(1);
        }

        byte[] data = Files.readAllBytes(Paths.get(args[0]));

        long start = System.nanoTime();

        long words = 0;
        boolean inWord = false;
        for (int i = 0; i < data.length; i++) {
            byte b = data[i];
            if (b == ' ' || b == '\t' || b == '\n' || b == '\r' || b == 0x0b || b == 0x0c) {
                inWord = false;
            } else {
                if (!inWord) {
                    words++;
                    inWord = true;
                }
            }
        }

        double ms = (System.nanoTime() - start) / 1e6;

        System.out.println("words: " + words);
        System.out.printf("RESULT:%.3f%n", ms);
    }
}
